### ISSUE TITLE
Upgrading from 3.0.0 to 3.3.1 changes severity to error (even for other analyzers)

### ISSUE THREAD

**Issue by Bouke** [dotnet/roslyn-analyzers#4398]
### Analyzer

Microsoft.CodeQuality.Analyzers

### Analyzer source

**NuGet Package**: https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers/3.3.1
**Version**: v3.3.1 (Latest)
**Visual Studio**: 2019 Version 16.7.6

### Describe the bug

After upgrading my solution to use the new v3.3.1 analyzers, the severity from various analyzers is incorrect. Restoring version 3.0.0 restores the severity. For example CA1031 is by default Warning and I haven't changed that. After upgrading to 3.3.1 it is now an Error. Only by specifying an override can I make it a Warning again.

The trigger condition is unclear to me.
- The solution on which I ran into this has custom rulesets, so I first tried migrating to editorconfig. This didn't solve the problem.
- I did a new checkout with the 3.3.1 versions. This appears to fix the issue. The code/ruleset is 100% identical. No editorconfig applies for both the old and new checkout.

### Steps To Reproduce

Create new project; e.g. CLI with .Net Framework. Install Microsoft.CodeQuality.Analyzers version 3.0.0. Write some code that triggers some warnings from this analyzer. Then upgrade to version 3.3.1. Notice how IntelliSense changes the severity of various warnings:

### Expected behavior

Upgrading from 3.0.0 shouldn't change severity (other than the default of the analyzer).

### Actual behavior

Upgrading to 3.0.0 changes the severity to Error of various Warnings, but it is unclear where this divergence is configured (if at all).

### Screenshots

With 3.0.0
![Screenshot 2020-11-01 at 15 04 16](https://user-images.githubusercontent.com/235882/97804987-8c801680-1c53-11eb-8158-f416f0fef230.png)

After upgrading to 3.3.1
![Screenshot 2020-11-01 at 15 05 29](https://user-images.githubusercontent.com/235882/97805005-b5081080-1c53-11eb-9a84-ce3cb4f674ce.png)

(Note that SXXX is coming from [SonarLint](https://marketplace.visualstudio.com/items?itemName=SonarSource.SonarLintforVisualStudio2019)).

**Comment by ShenValleyLewis** [comment]
This is happening for me as well, with Visual Studio Professional 2019, version 16.7.3.  I don't have the problem with the analyzer version 3.3.0 but I do have it with 3.3.1.

**Comment by mavasani** [comment]
@Bouke @ShenValleyLewis Can you try your scenarios without SonarLint package? Code quality analyzer package can only affect severity of CA diagnostics, so I am not sure how this could happen...

**Comment by ShenValleyLewis** [comment]
My projects are not using SonarLint, however they are using SonarAnalyzer.CSharp and StyleCop.Analyzers.  I uninstalled both of these other analyzers, but it doesn't affect the issue with Roslyn Analyzers.

**Comment by mavasani** [comment]
Just tried the below repro steps and I am unable to repro this:

1. Create a C# .NETFramework class library with below code:

```csharp
namespace N
{
    public class Class1
    {
        void M()
        {
            var a = 0;
        }
    }
}
```

2. Install Microsoft.CodeQualityAnalyzers version `3.0.0`

3. Confirm CA and CS intellisense warnings in error list:
```
Severity	Code	Description	Project	Source	File	Line	Suppression State
Warning	CS0219	The variable 'a' is assigned but its value is never used	ClassLibrary56	IntelliSense	Class1.cs	7	Active
Warning	CA1822	Member 'M' does not access instance data and can be marked as static	ClassLibrary56	IntelliSense	Class1.cs	5	Active

```

4. Upgrade to Microsoft.CodeQualityAnalyzers version `3.3.1`

5. Confirm exactly same set of warnings - no errors.

**Comment by mavasani** [comment]
> My projects are not using SonarLint, however they are using SonarAnalyzer.CSharp and StyleCop.Analyzers. I uninstalled both of these other analyzers, but it doesn't affect the issue with Roslyn Analyzers.

Can you please provide exact repro steps for your issue?

**Comment by ShenValleyLewis** [comment]
> > My projects are not using SonarLint, however they are using SonarAnalyzer.CSharp and StyleCop.Analyzers. I uninstalled both of these other analyzers, but it doesn't affect the issue with Roslyn Analyzers.
> 
> Can you please provide exact repro steps for your issue?

I don't know what additional information I can give, apart from what I have already stated.  If you need specific details about my Visual Studio version, Windows version, etc. just let me know.

**Comment by mavasani** [comment]
@ShenValleyLewis Can you please provide repro steps similar to the one mentioned in https://github.com/dotnet/roslyn-analyzers/issues/4398#issuecomment-720538539?

**Comment by ShenValleyLewis** [comment]
I created a new project with the following code:
`
namespace ClassLibrary1
{
    public class Class1
    {
        void Test()
        {
            int x = 0;
        }
    }
}
`

and the following NuGet packages:
Microsoft.CodeAnalysis.FxCopAnalyzers 3.3.0
SonarAnalyzer.CSharp 8.14.0.22654
StyleCop.Analyzers 1.1.118

I get the following warnings:
Warning	SA1633	The file header is missing or not located at the top of the file.
Warning	SA1517	Code should not contain blank lines at start of file
Warning	SA1600	Elements should be documented
Warning	CA1822	Member 'Test' does not access instance data and can be marked as static
Warning	SA1400	Element 'Test' should declare an access modifier
Warning	S1481	Remove the unused local variable 'x'.
Warning	IDE0059	Unnecessary assignment of a value to 'x'
Warning	CS0219	The variable 'x' is assigned but its value is never used

Then, I changed the Microsoft.CodeAnalysis.FxCopAnalyzers reference to 3.3.1 and now I get this:
Error	SA1633	The file header is missing or not located at the top of the file.
Error	SA1517	Code should not contain blank lines at start of file
Error	SA1600	Elements should be documented
Error	SA1400	Element 'Test' should declare an access modifier
Error	S1481	Remove the unused local variable 'x'.
Error	IDE0059	Unnecessary assignment of a value to 'x'
Error	CS0219	The variable 'x' is assigned but its value is never used
Warning	CA1822	Member 'Test' does not access instance data and can be marked as static

**Comment by JacobT14** [comment]
Can confirm this also happened to me.  I thought I was going crazy, but downgrading back to 3.3.0 puts all the warnings at the right severity.

**Comment by mavasani** [comment]
I am still unable to repro this:
![image](https://user-images.githubusercontent.com/10605811/97901695-5d1fe700-1cf1-11eb-9810-47d6cbf2c880.png)

Any chance you have an editorconfig entry `dotnet_analyzer_diagnostic.severity = error` somewhere in your repo that is bumping the default severity of all analyzer diagnostics?

**Comment by ShenValleyLewis** [comment]
> 
> 
> I am still unable to repro this:
> ![image](https://user-images.githubusercontent.com/10605811/97901695-5d1fe700-1cf1-11eb-9810-47d6cbf2c880.png)
> 
> Any chance you have an editorconfig entry `dotnet_analyzer_diagnostic.severity = error` somewhere in your repo that is bumping the default severity of all analyzer diagnostics?

My sample code was not created in any repository.  It was a brand new vanilla project/solution.  In any case, if that were the explanation, I would expect it to have the same effect whether using 3.3.0 or 3.3.1.

I can send you my complete project files, if that would help.  Also any global Visual Studio settings files, if you tell me what to look for.

**Comment by mavasani** [comment]
> I can send you my complete project files, if that would help.

@ShenValleyLewis That would be very helpful, thanks! Can you please also confirm the VS version you are using?

**Comment by mavasani** [comment]
One more piece of input - do you see these diagnostics as errors even on command line OR just while typing (intellisense)?

**Comment by ShenValleyLewis** [comment]
> 
> 
> One more piece of input - do you see these diagnostics as errors even on command line OR just while typing (intellisense)?

They appear as errors in the Error List window and also in the IDE (the red wiggly lines appear under the code).

**Comment by ShenValleyLewis** [comment]
> > I can send you my complete project files, if that would help.
> 
> @ShenValleyLewis That would be very helpful, thanks! Can you please also confirm the VS version you are using?

I was originally using 16.7.3 as stated in my original post.  I just upgraded to 16.7.7 but it hasn't made any difference.

[ClassLibrary1.zip](https://github.com/dotnet/roslyn-analyzers/files/5477250/ClassLibrary1.zip)

**Comment by mavasani** [comment]
> They appear as errors in the Error List window and also in the IDE (the red wiggly lines appear under the code).

Can you confirm the behavior if you build the project, either from command line or inside Visual Studio (for the latter look at the contents in the Output Window)?

**Comment by ShenValleyLewis** [comment]
> Can you confirm the behavior if you build the project, either from command line or inside Visual Studio (for the latter look at the contents in the Output Window)?

Here are the contents of the Output Window:

1>------ Build started: Project: ClassLibrary1, Configuration: Debug Any CPU ------
1>C:\code\swift-git\ClassLibrary1\ClassLibrary1\Class1.cs(8,17,8,18): warning CS0219: The variable 'x' is assigned but its value is never used
1>C:\code\swift-git\ClassLibrary1\ClassLibrary1\Class1.cs(1,1,1,1): warning SA1633: The file header is missing or not located at the top of the file.
1>C:\code\swift-git\ClassLibrary1\ClassLibrary1\Class1.cs(1,1,2,1): warning SA1517: Code should not contain blank lines at start of file
========== Build: 1 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========

**Comment by mavasani** [comment]
Thanks, I can repro it now - and this behavior seems to be specific to legacy csproj style .NET projects, it does not repro on SDK-style projects. I'll investigate.

**Comment by mavasani** [comment]
@Bouke @ShenValleyLewis So, I was able to repro the issue on 16.7.x builds, but this does not repro on following:

1. 16.8 Preview6 build: https://docs.microsoft.com/visualstudio/releases/2019/release-notes-preview
2. Internal 16.9 dogfood builds

So, it seems like there was some Roslyn/VS bug in 16.7 that seems to be causing it, which has been fixed in upcoming release.

**Comment by ShenValleyLewis** [comment]
> 
> So, it seems like there was some Roslyn/VS bug in 16.7 that seems to be causing it, which has been fixed in upcoming release.

Good to know!

**Comment by Bouke** [comment]
Thanks. We'll hold off upgrading the analyzers until the team has upgraded to 16.8. Maybe this warrants a note in the documentation somewhere?

**Comment by Ergamon** [comment]
Sorry for the duplicate. Seems I did not use the correct search terms ;)
